extra/src/main/java/aima/extra/probability/ProbabilityFactory.java

```diff
@@ -0,0 +1,91 @@
+package aima.extra.probability;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+
+/**
+ * Factory class to generate ProbabilityNumber types.
+ * 
+ * @author Nagaraj Poti
+ *
+ */
+public class ProbabilityFactory {
+
```

Comment:
Not sure this is how I'd approach the factory mechanism but lets experiment with for now.